src/Widgets/NotificationsList.vala

```diff
@@ -44,18 +43,15 @@ public class Notifications.NotificationsList : Gtk.ListBox {
     }
 
     public void add_entry (NotificationEntry entry) {
-        AppEntry? app_entry = null;
         if (entry.notification.app_info != null && entry.notification.app_info.get_id () != null) {
-            string[] desktop_id_list = {};
-            app_entries.foreach ((_app_entry) => {
-                var app_id = _app_entry.app_info.get_id ();
+            AppEntry? app_entry = null;
 
-                desktop_id_list += app_id;
-
-                if (app_id == entry.notification.desktop_id && app_entry == null) {
+            foreach (unowned AppEntry _app_entry in app_entries) {
+                if (_app_entry.app_info.get_id () == entry.notification.desktop_id) {
```

Comment:
It might be worth storing this value as `unowned string entry_desktop_id = entry.notification.desktop_id` before entering the loop :slightly_smiling_face: